app/controllers/products_controller.rb

```diff
@@ -1,2 +1,75 @@
 class ProductsController < ApplicationController
+
+  # Add controller filter once set up
+  # before_action :find_product, only: [:show, :edit, :update, :destroy?]
+
+  def index
+    @products = Product.all
+  end
+
+  def new
+    @product = Product.new
+  end
+
+  def create
```

Comment:
to link to a current user: 

`@current_merchant = Merchant.find_by(id: session[:user_id])`
`if @current_merchant.nil?`
` # or flash[:warning] and redirect somewhere
`   head :not_found`
`   return`
`end`